app/Providers/ValidationServiceProvider.php

```diff
@@ -83,6 +83,18 @@ public function boot()
                 return $count < 1;
             }
         });
+
+            Validator::extend('required_serial', function ($attribute, $value, $parameters, $validator) {
+                if (Setting::getSettings()->required_serial) {
+                    return !empty($value);
+                }
+
+                return true;
+            });
+            Validator::replacer('required_serial', function ($message, $attribute, $rule, $parameters) {
+                return str_replace(':attribute', $attribute, ':attribute is required.');
+            });
```

Comment:
Nitpick: these two blocks are indented more than the surrounding blocks. Doesn't need to hold up merging.